packages/cli/src/analytics/index.ts

```diff
@@ -0,0 +1,222 @@
+    // Check environment variable first
+    if (process.env.NO_MASTRA_TELEMETRY) {
+      return false;
+    }
+    // Default to enabled
+    return true;
+  }
+
+  private getDistinctId(): string {
+    // Use machine-id or generate a persistent ID
+    // This helps track unique CLI installations
+    const machineId = os.hostname();
+    return `mastra-${machineId}`;
+  }
+
+  private getSystemProperties(): Record<string, any> {
+    return {
+      os: process.platform,
+      os_version: os.release(),
+      node_version: process.version,
+      platform: process.arch,
+      session_id: this.sessionId,
+      cli_version: this.version || 'unknown',
+      machine_id: os.hostname(),
+    };
+  }
+
+  private captureSessionStart(): void {
+    if (!this.client) {
+      return;
+    }
+
+    this.client.capture({
+      distinctId: this.distinctId,
+      event: 'cli_session_start',
+      properties: {
+        ...this.getSystemProperties(),
+      },
+    });
+  }
+
+  trackCommand(options: {
+    command: string;
+    args?: Record<string, unknown>;
+    durationMs?: number;
+    status?: 'success' | 'error';
+    error?: string;
+  }): void {
+    try {
+      if (!this.client) {
+        return;
+      }
+
+      const commandData: CommandData = {
+        command: options.command,
+        status: options.status || 'success',
+      };
+
+      if (options.args) {
+        commandData.args = options.args;
+      }
+
+      if (options.durationMs) {
+        commandData.durationMs = options.durationMs;
+      }
+
+      if (options.error) {
+        commandData.error = options.error;
+      }
+
+      this.client.capture({
+        distinctId: this.distinctId,
+        event: 'cli_command',
+        properties: {
+          ...this.getSystemProperties(),
+          ...commandData,
+        },
+      });
+    } catch (e) {
+      //swallow
+    }
+  }
+
+  // Helper method to wrap command execution with timing
+  async trackCommandExecution<T>({
+    command,
+    args,
+    execution,
+  }: {
+    command: string;
+    args: Record<string, unknown>;
+    execution: () => Promise<T>;
+  }): Promise<T> {
+    const startTime = process.hrtime();
+
+    try {
+      const result = await execution();
+      const [seconds, nanoseconds] = process.hrtime(startTime);
+      const durationMs = seconds * 1000 + nanoseconds / 1000000;
+
+      this.trackCommand({
+        command,
+        args,
+        durationMs,
+        status: 'success',
+      });
+
+      return result;
+    } catch (error) {
+      const [seconds, nanoseconds] = process.hrtime(startTime);
+      const durationMs = seconds * 1000 + nanoseconds / 1000000;
+
+      this.trackCommand({
+        command,
+        args,
+        durationMs,
+        status: 'error',
+        error: error instanceof Error ? error.message : String(error),
+      });
+
+      throw error;
+    }
+  }
+
+  // Ensure PostHog client is shutdown properly
+  async shutdown(): Promise<void> {
```

Comment:
I noticed we call `shutdown()` after the --version command but not after the other commands? Should we call shutdown after the calls in `trackCommandExecution`?

Reply:
shutdown is happening on process.exit

ill remove it from version